lstchain/io/event_selection.py

```diff
@@ -186,16 +183,33 @@ def apply_global_theta_cut(self, data):
         return data[data["theta"].to_value(u.deg) < self.global_theta_cut]
 
     def energy_dependent_theta_cuts(
-        self, data, energy_bins, smoothing=None,
+        self,
+        data,
+        energy_bins,
+        use_same_disp_sign=True,
+        smoothing=None,
     ):
         """
-        Evaluating an optimized energy-dependent theta cuts, in a given
-        data, with provided reco energy bins, and other parameters to
-        pass to the pyirf.cuts.calculate_percentile_cut function.
+        Evaluating an optimized energy-dependent theta cuts, in a given MC data,
+        with provided reco energy bins, and other parameters to pass to the
+        pyirf.cuts.calculate_percentile_cut function.
+
+        For MC data, the reconstructed disp_sign may not be the same as the
+        True value, and thus resulting in a distinct angular distribution at
+        larger theta values owing to the events with incorrect disp_sign. This
+        affects the total range over which percentile cuts are evaluated. To
+        fix this by default we apply a mask on the data, so as to only use
+        events with the same disp_sign after reconstruction, for evaluating the
```

Comment:
I think the rationale for doing this is missing in this explanation.
Due to the events with wrongly reconstructed disp_sign, The PSF is bi-modal, and what we typically want is to keep a given fraction of the central peak.